spec/models/reducer_spec.rb

```diff
@@ -41,6 +41,19 @@ def reduce_into(extracts, reductions=nil)
     klass.new
   end
 
+  it 'does not try to reduce empty extract sets' do
+    allow_any_instance_of(ExtractFetcher).to receive(:extracts).and_return([])
+
+    reduction_fetcher = instance_double(ReductionFetcher, retrieve: SubjectReduction)
+    expect(ReductionFetcher).to receive(:new).and_return(reduction_fetcher)
```

Comment:
It seems like this is more testing that the spec is being set up correctly than any specific behavior...is this necessary, or maybe it makes more sense for this to be in a separate block if it's testing something specific?

Reply:
It's testing specifically `reducer.rb:62` but there's a lot of faff necessary to build the parameters that `Reducer#process` expects. This really was the simplest failing test I could write.

Reply:
I will mark this resolved and merge for now so I can continue testing BUT if you have further concerns don't hesitate to let me know and we can talk more about the test.

Reply:
There's a lot of that in this repo, no worries. More curious about explicitly the `expect(ReductionFetcher)` bit since it doesn't build anything and I wasn't sure what it was asking.